--- toolkit/components/extensions/webrequest/WebRequestUpload.sys.mjs.orig	2026-05-15 21:52:15
+++ toolkit/components/extensions/webrequest/WebRequestUpload.sys.mjs	2026-05-15 21:52:15
@@ -532,6 +532,11 @@
     }
 
     try {
+      channel.QueryInterface(Ci.nsIHttpChannel).getRequestHeader("X-Berytus-Channel-Id");
+      return { error: "Upload streams with Berytus Encrypted Packet(s) are unsupported" };
+    } catch (e) { }
+
+    try {
       let stream = channel.uploadStream;
 
       let formData = createFormData(stream, channel);
