--- dom/base/FormData.h.orig	2026-05-15 21:51:47
+++ dom/base/FormData.h	2026-05-15 21:52:14
@@ -25,6 +25,8 @@
 class HTMLFormElement;
 class GlobalObject;
 
+class BerytusEncryptedPacket;
+
 class FormData final : public nsISupports,
                        public HTMLFormSubmission,
                        public nsWrapperCache {
@@ -47,6 +49,8 @@
 
   void SetNameFilePair(FormDataTuple* aData, const nsAString& aName,
                        File* aFile);
+  void SetNameBlobPair(FormDataTuple* aData, const nsAString& aName,
+                   Blob* aBlob);
 
   void SetNameDirectoryPair(FormDataTuple* aData, const nsAString& aName,
                             Directory* aDirectory);
@@ -167,6 +171,8 @@
   RefPtr<Element> mSubmitter;
 
   nsTArray<FormDataTuple> mFormData;
+
+  friend class BerytusEncryptedPacket;
 };
 
 }  // namespace dom
