--- toolkit/components/extensions/jar.mn.orig	2025-01-29 21:06:25
+++ toolkit/components/extensions/jar.mn	2025-01-29 21:06:25
@@ -10,6 +10,7 @@
     content/extensions/parent/ext-activityLog.js (parent/ext-activityLog.js)
     content/extensions/parent/ext-alarms.js (parent/ext-alarms.js)
     content/extensions/parent/ext-backgroundPage.js (parent/ext-backgroundPage.js)
+    content/extensions/parent/ext-berytus.js (parent/ext-berytus.js)
     content/extensions/parent/ext-browserSettings.js (parent/ext-browserSettings.js)
     content/extensions/parent/ext-browsingData.js (parent/ext-browsingData.js)
 #ifndef ANDROID
@@ -49,6 +50,7 @@
     content/extensions/parent/ext-webRequest.js (parent/ext-webRequest.js)
     content/extensions/parent/ext-webNavigation.js (parent/ext-webNavigation.js)
     content/extensions/child/ext-backgroundPage.js (child/ext-backgroundPage.js)
+    content/extensions/child/ext-berytus.js (child/ext-berytus.js)
     content/extensions/child/ext-contentScripts.js (child/ext-contentScripts.js)
     content/extensions/child/ext-declarativeNetRequest.js (child/ext-declarativeNetRequest.js)
     content/extensions/child/ext-extension.js (child/ext-extension.js)
