--- toolkit/components/extensions/schemas/jar.mn.orig	2025-04-10 17:31:15
+++ toolkit/components/extensions/schemas/jar.mn	2025-04-10 17:31:15
@@ -6,6 +6,7 @@
 % content extensions %content/extensions/
     content/extensions/schemas/activity_log.json
     content/extensions/schemas/alarms.json
+    content/extensions/schemas/berytus.json
     content/extensions/schemas/browser_action.json
     content/extensions/schemas/browser_settings.json
     content/extensions/schemas/browsing_data.json
