--- ipc/glue/BackgroundParentImpl.h.orig	2026-05-15 20:07:22
+++ ipc/glue/BackgroundParentImpl.h	2026-05-15 21:52:15
@@ -358,6 +358,9 @@
       NotNull<nsIPrincipal*> aPrincipal, const Maybe<nsID>& aClientId) final;
 
   already_AddRefed<PFetchParent> AllocPFetchParent() override;
+
+  virtual mozilla::ipc::IPCResult RecvCreateMaskManagerParent(
+      Endpoint<::mozilla::berytus::PMaskManagerParent>&& aEndpoint) override;
 };
 
 }  // namespace mozilla::ipc
