--- ipc/glue/BackgroundParentImpl.h.orig	2025-10-31 19:44:55
+++ ipc/glue/BackgroundParentImpl.h	2025-10-31 19:45:39
@@ -351,6 +351,9 @@
       NotNull<nsIPrincipal*> aPrincipal, const Maybe<nsID>& aClientId) final;
 
   already_AddRefed<PFetchParent> AllocPFetchParent() override;
+
+  virtual mozilla::ipc::IPCResult RecvCreateMaskManagerParent(
+      Endpoint<::mozilla::berytus::PMaskManagerParent>&& aEndpoint) override;
 };
 
 }  // namespace mozilla::ipc
