--- browser/base/content/navigator-toolbox.inc.xhtml.orig	2024-10-31 21:28:22
+++ browser/base/content/navigator-toolbox.inc.xhtml	2024-10-31 21:28:22
@@ -283,6 +283,8 @@
                        data-l10n-id="urlbar-web-authn-anchor"/>
                 <image id="identity-credential-notification-icon" class="notification-anchor-icon" role="button"
                       data-l10n-id="identity-credential-urlbar-anchor"/>
+                <image id="berytus-notification-icon" class="notification-anchor-icon" role="button"
+                       alt="Open berytus panel"/>
                 <image id="storage-access-notification-icon" class="notification-anchor-icon storage-access-icon" role="button"
                        data-l10n-id="urlbar-storage-access-anchor"/>
               </box>
