--- toolkit/components/extensions/child/ext-toolkit.js.orig	2025-04-10 17:31:15
+++ toolkit/components/extensions/child/ext-toolkit.js	2025-04-10 17:31:15
@@ -16,6 +16,11 @@
       ["runtime", "getBackgroundPage"],
     ],
   },
+  berytus: {
+    url: "chrome://extensions/content/child/ext-berytus.js",
+    scopes: ["addon_child"],
+    paths: [["berytus"]],
+  },
   contentScripts: {
     url: "chrome://extensions/content/child/ext-contentScripts.js",
     scopes: ["addon_child"],
