--- dom/file/Blob.h.orig	2026-05-15 21:51:47
+++ dom/file/Blob.h	2026-05-15 21:52:14
@@ -134,6 +134,9 @@
   virtual ~Blob();
 
   virtual bool HasFileInterface() const { return false; }
+  virtual bool HasBerytusEncryptedPacketInterface() const { return false; }
+  template <typename T>
+  friend T* TryDowncastBlob(Blob* aBlob);
 
   already_AddRefed<Promise> ConsumeBody(BodyConsumer::ConsumeType aConsumeType,
                                         ErrorResult& aRv) const;
