--- dom/fetch/FetchDriver.h.orig	2026-05-15 21:51:47
+++ dom/fetch/FetchDriver.h	2026-05-15 21:52:14
@@ -36,6 +36,8 @@
 
 namespace dom {
 
+#define NS_FETCH_DRIVER_HTTP_FETCH_TOPIC "fetch-driver-http-fetch"
+
 class Document;
 class InternalRequest;
 class InternalResponse;
@@ -154,6 +156,9 @@
     mIsOn3PCBExceptionList = aIsOn3PCBExceptionList;
   }
 
+  SafeRefPtr<InternalRequest> GetRequest() const;
+  bool GetChannel(nsIChannel** aChannel) const;
+
  private:
   nsCOMPtr<nsIPrincipal> mPrincipal;
   nsCOMPtr<nsILoadGroup> mLoadGroup;
@@ -242,6 +247,8 @@
                          bool aStripAuthHeader) const;
 
   void FinishOnStopRequest(AlternativeDataStreamListener* aAltDataListener);
+
+  nsresult NotifyHttpFetchObservers(nsCOMPtr<nsIChannel>& aChannel);
 };
 
 }  // namespace dom
