--- dom/fetch/FetchDriver.h.orig	2025-10-15 22:30:35
+++ dom/fetch/FetchDriver.h	2025-10-23 21:32:09
@@ -35,6 +35,8 @@
 
 namespace dom {
 
+#define NS_FETCH_DRIVER_HTTP_FETCH_TOPIC "fetch-driver-http-fetch"
+
 class Document;
 class InternalRequest;
 class InternalResponse;
@@ -149,6 +151,9 @@
     mIsThirdPartyWorker = aIsThirdPartyWorker;
   }
 
+  void GetRequest(InternalRequest** aRequest) const;
+  bool GetChannel(nsIChannel** aChannel) const;
+
  private:
   nsCOMPtr<nsIPrincipal> mPrincipal;
   nsCOMPtr<nsILoadGroup> mLoadGroup;
@@ -231,6 +236,8 @@
                          bool aStripAuthHeader) const;
 
   void FinishOnStopRequest(AlternativeDataStreamListener* aAltDataListener);
+
+  nsresult NotifyHttpFetchObservers(nsCOMPtr<nsIChannel>& aChannel);
 };
 
 }  // namespace dom
