--- dom/fetch/Request.cpp.orig	2025-09-20 16:11:07
+++ dom/fetch/Request.cpp	2025-10-23 23:04:50
@@ -7,10 +7,14 @@
 #include "Request.h"
 
 #include "js/Value.h"
+#include "mozilla/dom/BerytusEncryptedPacket.h"
+#include "nsDebug.h"
+#include "nsIObserver.h"
+#include "nsISupports.h"
 #include "nsIURI.h"
 #include "nsNetUtil.h"
 #include "nsPIDOMWindow.h"
-
+#include "nsIOService.h"
 #include "mozilla/ErrorResult.h"
 #include "mozilla/StaticPrefs_network.h"
 #include "mozilla/dom/Headers.h"
@@ -27,6 +31,8 @@
 #include "mozilla/dom/ReadableStreamDefaultReader.h"
 
 namespace mozilla::dom {
+
+NS_IMPL_ISUPPORTS(Request::ConstructorNotification, nsISupports)
 
 NS_IMPL_ADDREF_INHERITED(Request, FetchBody<Request>)
 NS_IMPL_RELEASE_INHERITED(Request, FetchBody<Request>)
@@ -69,7 +75,7 @@
 
 Request::~Request() = default;
 
-SafeRefPtr<InternalRequest> Request::GetInternalRequest() {
+SafeRefPtr<InternalRequest> Request::GetInternalRequest() const {
   return mRequest.clonePtr();
 }
 
@@ -470,7 +476,55 @@
       }
     }
   }
+  NotifyConstructorObservers(domRequest, aInit);
   return domRequest;
+}
+
+nsresult Request::NotifyConstructorObservers(
+    SafeRefPtr<Request>& aRequest,
+    const RequestInit& aInit) {
+  NS_ENSURE_TRUE(NS_IsMainThread(), NS_ERROR_NOT_SAME_THREAD);
+  nsresult rv;
+  RefPtr<Request::ConstructorNotification> notif =
+    new Request::ConstructorNotification(
+      aRequest.unsafeGetRawPtr(), &aInit);
+  nsCOMPtr<nsIObserverService> obs = services::GetObserverService();
+  if (NS_WARN_IF(!obs)) {
+    return NS_ERROR_FAILURE;
+  }
+  nsCOMPtr<nsISupports> asSupports = do_QueryInterface(notif);
+  if (NS_WARN_IF(!asSupports)) {
+    return NS_ERROR_FAILURE;
+  }
+  // TODO(berytus): Remove the below block.
+  {
+    nsCOMPtr<nsISimpleEnumerator> observers;
+    rv = obs->EnumerateObservers(NS_FETCH_REQUEST_CONSTRUCTOR_TOPIC, getter_AddRefs(observers));
+    NS_ENSURE_SUCCESS(rv, rv);
+    NS_ENSURE_TRUE(observers, NS_ERROR_FAILURE);
+    bool hasMore = false;
+    rv = observers->HasMoreElements(&hasMore);
+    NS_ENSURE_SUCCESS(rv, rv);
+    if (!hasMore) {
+      printf("No Observers registered for %s\n", NS_FETCH_REQUEST_CONSTRUCTOR_TOPIC);
+      return NS_OK;
+    }
+    while (hasMore) {
+      nsCOMPtr<nsISupports> observer;
+      rv = observers->GetNext(getter_AddRefs(observer));
+      NS_ENSURE_SUCCESS(rv, rv);
+      NS_ENSURE_TRUE(observer, NS_ERROR_FAILURE);
+      printf("TopicObserver(%s, %p)\n", NS_FETCH_REQUEST_CONSTRUCTOR_TOPIC, observer.get());
+      rv = observers->HasMoreElements(&hasMore);
+      NS_ENSURE_SUCCESS(rv, rv);
+    }
+  }
+  rv = obs->NotifyObservers(asSupports,
+                            NS_FETCH_REQUEST_CONSTRUCTOR_TOPIC,
+                            nullptr);
+  NS_ENSURE_SUCCESS(rv, rv);
+  notif->MarkAsUnsafe();
+  return NS_OK;
 }
 
 SafeRefPtr<Request> Request::Clone(ErrorResult& aRv) {
