--- dom/fetch/Request.cpp.orig	2025-09-20 16:11:07
+++ dom/fetch/Request.cpp	2025-10-19 14:41:20
@@ -7,10 +7,12 @@
 #include "Request.h"
 
 #include "js/Value.h"
+#include "mozilla/dom/BerytusEncryptedPacket.h"
+#include "nsISupports.h"
 #include "nsIURI.h"
 #include "nsNetUtil.h"
 #include "nsPIDOMWindow.h"
-
+#include "nsIOService.h"
 #include "mozilla/ErrorResult.h"
 #include "mozilla/StaticPrefs_network.h"
 #include "mozilla/dom/Headers.h"
@@ -27,6 +29,8 @@
 #include "mozilla/dom/ReadableStreamDefaultReader.h"
 
 namespace mozilla::dom {
+
+NS_IMPL_ISUPPORTS(Request::ConstructorNotification, nsISupports)
 
 NS_IMPL_ADDREF_INHERITED(Request, FetchBody<Request>)
 NS_IMPL_RELEASE_INHERITED(Request, FetchBody<Request>)
@@ -69,7 +73,7 @@
 
 Request::~Request() = default;
 
-SafeRefPtr<InternalRequest> Request::GetInternalRequest() {
+SafeRefPtr<InternalRequest> Request::GetInternalRequest() const {
   return mRequest.clonePtr();
 }
 
@@ -470,9 +474,36 @@
       }
     }
   }
+  aRv = NotifyConstructorObservers(domRequest, aInit);
+  if (NS_WARN_IF(aRv.Failed())) {
+    return nullptr;
+  }
   return domRequest;
 }
 
+nsresult Request::NotifyConstructorObservers(
+    SafeRefPtr<Request>& aRequest,
+    const RequestInit& aInit) {
+
+  RefPtr<Request::ConstructorNotification> notif =
+    new Request::ConstructorNotification(
+      aRequest.unsafeGetRawPtr(), &aInit);
+  nsCOMPtr<nsIObserverService> obs =
+      static_cast<nsIObserverService*>(net::gIOService);
+  if (NS_WARN_IF(!obs)) {
+    return NS_ERROR_FAILURE;
+  }
+  nsCOMPtr<nsISupports> asSupports = do_QueryInterface(notif);
+  if (NS_WARN_IF(!asSupports)) {
+    return NS_ERROR_FAILURE;
+  }
+  obs->NotifyObservers(asSupports,
+                       NS_FETCH_REQUEST_CONSTRUCTOR_TOPIC,
+                       nullptr);
+  notif->MarkAsUnsafe();
+  return NS_OK;
+}
+
 SafeRefPtr<Request> Request::Clone(ErrorResult& aRv) {
   if (BodyUsed()) {
     aRv.ThrowTypeError<MSG_FETCH_BODY_CONSUMED_ERROR>();
