--- dom/webidl/Navigator.webidl.orig	2026-05-15 21:51:47
+++ dom/webidl/Navigator.webidl	2026-05-15 21:52:14
@@ -281,6 +281,11 @@
 };
 
 partial interface Navigator {
+  [SameObject]
+  readonly attribute BerytusChannelContainer berytusChannel;
+};
+
+partial interface Navigator {
   [Throws, Pref="beacon.enabled"]
   boolean sendBeacon(DOMString url,
                      optional BodyInit? data = null);
